shell/common/shell.cc

```diff
@@ -682,6 +682,16 @@ void Shell::OnPlatformViewSetViewportMetrics(const ViewportMetrics& metrics) {
   FML_DCHECK(is_setup_);
   FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());
 
+  // This is the formula Android uses.
+  // https://android.googlesource.com/platform/frameworks/base/+/master/libs/hwui/renderthread/CacheManager.cpp#41
+  size_t max_bytes = metrics.physical_width * metrics.physical_height * 12 * 4;
```

Comment:
You can move this to a constant in rasterizer and use `std::optional`s default as discussed above.

Reply:
I'm not sure I'm clear on which part should be the constant.  Just the `12 * 4` part?

Reply:
maybe a static method on rasterizer that takes `width` and `height`?